otherlibs/dynlink/nodynlink.ml

```diff
@@ -19,37 +19,60 @@
 [@@@ocaml.warning "+a-4-30-40-41-42"]
 
 module DC = Dynlink_common
+module DT = Dynlink_types
 
-let not_available () =
+let not_available _ =
   failwith "No support for native dynlink on this platform"
 
 module Not_available = struct
   module Unit_header = struct
     type t = unit
 
-    let defines _ = not_available ()
-    let unsafe_module _ = not_available ()
+    let name = not_available
+    let crc = not_available
+
+    let interface_imports = not_available
+    let implementation_imports = not_available
+
+    let defined_symbols = not_available
+    let unsafe_module = not_available
   end
 
   type handle = unit
 
-  let default_crcs = ref []
-
-  let init () = not_available ()
+  let init = not_available
 
   let is_native = false
-  let adapt_filename f = f
+  let adapt_filename = not_available
 
-  let iter_initial_units _ = ()
+  let num_globals_inited = not_available
+
+  let fold_initial_units ~init ~f:_ = not_available init
```

Comment:
The only functional difference with @mshinwell's patch is that `adapt_filename`, `num_globals_inited` and `fold_initial_units` raise `Failure` here instead of returning dummy values. Unless Mark has an objection, I think that's OK.